triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go

```diff
@@ -27,16 +27,20 @@ func (e *ConnectionError) Error() string {
 	return e.Message
 }
 
+type NatsConn interface {
```

Comment:
This seems like a generic interface for publishing but is restricted by its name to NatsConn. Maybe just Publisher?

Reply:
This interface is required in order to mock the behaviour of the nats conn. The real NatsConn isn't an interface so I'm relying on golang's duck typing to get enable the mocking.
`NatsPublisher.Publish` calls `NatsConn.Publish`.

Reply:
https://github.com/sky-uk/kfp-operator/pull/419/commits/5235d4e57572834b8bdf681de6bc3dd247e7f2ea
I've made it lowercase for the moment so nothing outside can implement it

Reply:
I think if we do end up with multiple implementations of the `PublisherHandler`, they will split into different packages, with `natsConn` moving into the nats-only package.